tool/src/java/org/sakaiproject/gradebookng/tool/pages/GradebookPage.java

```diff
@@ -261,8 +266,22 @@ public void populateItem(Item cellItem, String componentId, IModel rowModel) {
                 	courseGrade = getString("label.coursegrade.nopermission");
                 }
 
-                cellItem.add(new Label(componentId, courseGrade));
-                cellItem.setOutputMarkupId(true);
+                Label courseGradeLabel = new Label(componentId, Model.of(courseGrade)) {
+                    @Override
+                    public void onEvent(IEvent<?> event) {
+                        super.onEvent(event);
+                        if (event.getPayload() instanceof ScoreChangedEvent) {
+                            ScoreChangedEvent scoreChangedEvent = (ScoreChangedEvent) event.getPayload();
+                            if (studentGradeInfo.getStudentUuid().equals(scoreChangedEvent.getStudentUuid())) {
+                                CourseGrade courseGrade = businessService.getCourseGrade(scoreChangedEvent.getStudentUuid());
```

Comment:
Sounds good to me. Could you collect everything you need from this and raise a ticket for me?